src/MPAGSCipher/CaesarCipher.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef MPAGSCIPHER_CAESARCIPHER_HPP
+#define MPAGSCIPHER_CAESARCIPHER_HPP
+
+#include "CipherMode.hpp"
+#include <string>
+#include <vector>
+
+class CaesarCipher {
+    public:
+    explicit CaesarCipher(const std::size_t& key);
+
+    explicit CaesarCipher(const std::string& key);
+
+    std::string applyCipher (const std::string& inputText, const CipherMode encrypt) const;
+
+    private:
+
+    std::size_t key_ ;
+    const std::vector<char> alphabet_;
```

Comment:
Couple of quick observations:
* You can give these both default values, which makes writing the constructors more lightweight
* Best to swap the order of these declarations - this then allows you to use the size of `alphabet_` to truncate `key_` at construction time
```suggestion
    const std::vector<char> alphabet_{'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'};
    std::size_t key_ {0};
```